### PR TITLE
ci: Flesh out preview and release workflows

### LINKED ISSUE #13: Flesh out Preview and Release workflows
## Description

Issue #2 is going to be used to create the basic shell for the `Preview` and `Release` workflows. That is, they will be no-ops, but with the `workflow_dispatch` trigger set to enable iteration within a branch to get them where they need to be. This issue is meant to flesh out these workflows to their desired end state.

## Additional Details

* The `Preview` workflow will be used to
  * create a dev version and related package artifacts
  * optionally upload the package artifacts to TestPyPI
  * enable Staging-style testing before pushing to production
* The `Release` workflow will be used to
  * create a release version and related package artifacts
  * create a GitHub release with the package artifacts
    * Ensure the change log is updated and included in the release
  * upload the package artifacts to PyPI
* Consider using repository environments
  * for separation of secrets
  * for enabling manual approval, by group, of release and deployment actions 

## Acceptance Criteria

- [ ] `Preview` workflow is available for manual and automatic usage, with features outlined above
- [ ] `Release` workflow is available for automatic usage, with features outlined above
- [ ] Documentation is updated

### PR DESCRIPTION
**Pull request by maxrake**
# Overview

This PR Implements the preview and release workflows for the repository, based on the [GitFlow workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) currently defined in the [Phylum Development Standards and Expectations](https://www.notion.so/phylum/Development-Standards-and-Expectations-07f01c12f56b4bc099840d6074c92615). This may not be the best way forward for a public Python project that doesn't really follow the Staging/Production model...but that discussion can take place in other channels and with new issues.

The documentation was split out of a single README and does a better job than I can here in this summary of explaining the process of contributing, releasing, and how the workflows are setup. It might be best to start with reviewing the docs.

* Relevant links
  * [Release Workflow History](https://github.com/phylum-dev/phylum-ci/actions/workflows/release.yml)
  * [GitHub Releases](https://github.com/phylum-dev/phylum-ci/releases)
  * [Example of what a successful deployment looks like](https://github.com/phylum-dev/phylum-ci/actions/runs/2106375149)
  * [Release History on TestPyPI](https://test.pypi.org/project/phylum-ci/#history)
  * [Release History on PyPI](https://pypi.org/project/phylum-ci/#history)
  * [repository settings for `Environments`](https://github.com/phylum-dev/phylum-ci/settings/environments)

## Checklist

- [x] Does this PR have an associated issue?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [x] Have you created sufficient tests?
- [x] Have you updated all affected documentation?
- [x] Have you updated the CHANGELOG?

## Issue

Closes #13

### CODE REVIEW

**Review by cd-work — COMMENTED** [review]
Since I'm completely unfamiliar with the release process, I can't really provide an in-depth review. But I've given it a quick read to ensure everything makes sense.

File: README.md [hunk 1]
```diff
@@ -9,67 +9,48 @@
...
-    ```sh
-    cd phylum-ci
-    poetry install
-    ```
+It requires Python 3.7+ to run.
 
-5. Create a branch for local development:
+### Usage
+<!-- TODO: Fill this section in as functionality is added -->
+The `phylum-ci` package exposes it's functionality with a command line interface (CLI). To view the options available
```
[the remark this answers is not in the document]
Comment (maxrake) [reply README.md]: accepted change

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# Contributing
```
[the remark this answers is not in the document]
Comment (maxrake) [reply docs/contributing.md]: Thanks...I'm still new to the GitHub ecosystem and learning all these standards. The change was made.

**Review by cd-work — APPROVED** [review]

**Comment by maxrake** [comment]
The integrations team agreed it was okay to merge this PR now, even though #20 is coming soon to replace this process.

**Pull request merged**